vue-app/src/plugins/Web3/connectors/WalletConnectConnector.ts

```diff
@@ -0,0 +1,37 @@
+import WalletConnectProvider from '@walletconnect/web3-provider'
+
+export default {
+  // TODO: add better return type
+  connect: async (): Promise<any | undefined> => {
+    const provider = new WalletConnectProvider({
+      infuraId: process.env.VUE_APP_INFURA_ID,
+      rpc: {
+        [process.env.VUE_APP_ETHEREUM_API_CHAINID!]:
+          process.env.VUE_APP_ETHEREUM_API_URL!,
+      },
```

Comment:
By design WC fallsback to infura, that is why we have the `infuraId` option. To avoid that behavior, you have the `rpc` option where you define each chain you want to override or support. In case you don't want to use infura at all, you would need to override all the chains supported by infura. The list is here: https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/helpers/utils/src/constants.ts#L28

So, for a given connection it will take the chainId the user's wallet is using and will look for a defined node:
1. First, it will check if there is any node for that chain defined in the `rpc` option.
2. If not, will use an infura node for that chain.
3. If not, it throws an error.

All this behavior is because we are using the abstraction `web3-provider`. If we would need more control/flexibility (e.g. not giving an `infuraId` and only support n numbers of chains), we should use the client https://docs.walletconnect.org/quick-start/dapps/client but that is an extra effort I think we don't need right now. Let me know if you think differently and we might estimate how much it is.

Reply:
If we decide to not use infura and go with alchemy, we would need to do the following:
```
const provider = new WalletConnectProvider({
  rpc: {
    1: "https://eth-mainnet.alchemyapi.io/v2/{key}",
    3: "https://eth-ropsten.alchemyapi.io/v2/{key}",
    4: "https://eth-rinkeby.alchemyapi.io/v2/{key}",
    5: "https://eth-goerli.alchemyapi.io/v2/{key}",
    42: "https://eth-kovan.alchemyapi.io/v2/{key}",
  },
});
```

Reply:
I agree I'm ok with not going through this extra effort just to avoid reliance on Infura. We still need to use a node endpoint & this implementation still allows us to specify which we want to use (e.g. Alchemy or our own node). I don't see harm in adding Infura as a backup.